src/Control/Monad/Logic/Sequence/Internal.hs

```diff
@@ -284,25 +284,25 @@ observeT (toView -> m) = m >>= go where
   go Empty = fail "No results."
 {-# INLINE observeT #-}
 
-observe :: Seq a -> a
+observe :: Seq a -> Maybe a
 observe (toView -> m) = case runIdentity m of
-  a :< _ -> a
-  Empty -> error "No results."
+  a :< _ -> Just a
+  Empty -> Nothing
 {-# INLINE observe #-}
 
-observeMaybeT :: Monad m => SeqT m (Maybe a) -> m (Maybe a)
-observeMaybeT (toView -> m) = m >>= go where
+observeTMaybe :: Monad m => SeqT m (Maybe a) -> m (Maybe a)
```

Comment:
Is this really the right type? I would really want

```haskell
observeTMaybe :: Monad m => SeqT m a -> m (Maybe a)
```

as a way to `observeT` is a non-`MonadFail` monad. Then your version can be implemented inline quite cleanly:

```haskell
yourObserveMaybeT s = myObserveMaybeT $ s >>= maybe empty pure
```